src/openms/include/OpenMS/METADATA/ID/IdentificationData.h

```diff
@@ -0,0 +1,556 @@
+
+    using PeakAnnotations = IdentificationDataInternal::PeakAnnotations;
+    using IdentifiedMoleculeRef =
+      IdentificationDataInternal::IdentifiedMoleculeRef;
+
+    using MoleculeQueryMatch = IdentificationDataInternal::MoleculeQueryMatch;
+    using MoleculeQueryMatches =
+      IdentificationDataInternal::MoleculeQueryMatches;
+    using QueryMatchRef = IdentificationDataInternal::QueryMatchRef;
+
+    // @TODO: allow multiple sets of groups, like with parent molecules
+    // ("ParentMoleculeGroupings")?
+    using QueryMatchGroup = IdentificationDataInternal::QueryMatchGroup;
+    using QueryMatchGroups = IdentificationDataInternal::QueryMatchGroups;
+    using MatchGroupRef = IdentificationDataInternal::MatchGroupRef;
+
+    using ParentMoleculeGroup = IdentificationDataInternal::ParentMoleculeGroup;
+    using ParentMoleculeGroups =
+      IdentificationDataInternal::ParentMoleculeGroups;
+    using ParentGroupRef = IdentificationDataInternal::ParentGroupRef;
+    using ParentMoleculeGrouping =
+      IdentificationDataInternal::ParentMoleculeGrouping;
+    using ParentMoleculeGroupings =
+      IdentificationDataInternal::ParentMoleculeGroupings;
+
+    using AddressLookup = boost::unordered_set<uintptr_t>;
+
+
+    /// Default constructor
+    IdentificationData():
+      current_step_ref_(processing_steps_.end())
+    {
+    }
+
+    // Copy constructor - not allowed, as references would be invalidated:
+    // @TODO: implement using deep copy
+    IdentificationData(const IdentificationData& other) = delete;
+
+    /// Move constructor
+    IdentificationData(IdentificationData&& other):
+      current_step_ref_(other.current_step_ref_)
+    {
+      input_files_.swap(other.input_files_);
+      processing_software_.swap(other.processing_software_);
+      processing_steps_.swap(other.processing_steps_);
+      db_search_params_.swap(other.db_search_params_);
+      db_search_steps_.swap(other.db_search_steps_);
+      score_types_.swap(other.score_types_);
+      data_queries_.swap(other.data_queries_);
+      parent_molecules_.swap(other.parent_molecules_);
+      parent_molecule_groupings_.swap(other.parent_molecule_groupings_);
+      identified_peptides_.swap(other.identified_peptides_);
+      identified_compounds_.swap(other.identified_compounds_);
+      identified_oligos_.swap(other.identified_oligos_);
+      query_matches_.swap(other.query_matches_);
+      query_match_groups_.swap(other.query_match_groups_);
+      swap(current_step_ref_, other.current_step_ref_);
+      // look-up tables:
+      data_query_lookup_.swap(other.data_query_lookup_);
+      parent_molecule_lookup_.swap(other.parent_molecule_lookup_);
+      identified_peptide_lookup_.swap(other.identified_peptide_lookup_);
+      identified_compound_lookup_.swap(other.identified_compound_lookup_);
+      identified_oligo_lookup_.swap(other.identified_oligo_lookup_);
+      query_match_lookup_.swap(other.query_match_lookup_);
```

Comment:
I think the default should be `mvar( std::move(other.mvar) )` for all member variables `mvar` unless you have a good reason - `std::swap` will actually create a temporary object and is extra work.

https://stackoverflow.com/questions/9456910/how-to-define-a-move-constructor

Reply:
The implementation has been replaced by `= default`.

Reply:
Apparently the `= default` non-implementation didn't work on some platform (Windows or Mac?), so I've made the changes that Hannes suggested.